pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -163,17 +163,21 @@
   @GET
   @Produces(MediaType.APPLICATION_JSON)
   @Path("/segments/{tableName}")
-  @ApiOperation(value = "List all segments", notes = "List all segments")
+  @ApiOperation(value = "List all segments. The 'liveOnly' parameter is used to get the list of segments"
```

Comment:
All the internal implementation can be fixed, let us make sure we get the parameter name right. Just suggesting here: How about "includeIgnoredSegments"? You can write an explanation that segments are ignored when the lineage information indicates that they are superseded by newer segments.

Reply:
There can be multiple kinds of segments that should be ignored. E.g. like Jackie mentioned, segments in ERROR state or not served by servers can also be ignored. Plus, the existing behavior should the default behavior, which means that  even if we introduce this `includeIgnoredSegments`, the value should be false by default.

How about renaming it to `queryableSegments` and make it false by default?

Reply:
Discussed it offline. The existing listSegments returns all the segments regardless of their states. Thus, the optional parameter should be called sth like `excludeReplacedSegments` to indicate that replaced segments are filtered out in the response payload. Later on, we can add more and more parameters like `excludeErrorSegments`, `excludeConsumingSegments`  to filter more segments.